### PR TITLE
Move generic tests from burn-flex to burn-backend-tests

### PR DESCRIPTION
**Pull request by antimora**
Addresses #4774.

Moves backend-agnostic tests out of `burn-flex/src/ops/` into `burn-backend-tests`, rewrites the remaining flex-specific tests to drop `burn-tensor`, and flags backend semantic divergences as tracked issues.

### Rule of thumb

1. Test uses only public API + `.flip()`/`.transpose()`/`.narrow()`/`.permute()`/`.swap_dims()` → move to **backend-tests** (runs on every backend).
2. Test probes flex internals (SIMD/rayon boundaries, `binary_op`, `scalar_op`, `broadcast_attn_mask_bias`, `Flex::q_*`, dtype storage like `IntDType::U8`, stride metadata) → keep in **flex**.
3. Test surfaces a semantic divergence between backends → keep in **backend-tests** behind a `#[cfg]` gate citing a tracking issue. Failures are signals.

### Migrations

| Category | Moved to backend-tests |
|---|---|
| Stride-through-op | softmax non-contig · add/sub/mul/scalar transposed+narrowed · 9 bool flipped · 8 comparison stride · 7 matmul transposed · 7 mask stride · 20 reduce (9 agg + 8 arg + 3 max/min NaN) · 8 unary stride · 8 expand stride · 3 cumulative NaN · 1 gather/scatter empty-indices |
| Dropped as redundant | ~145 tests across conv, conv_transpose, reduce, matmul, pool, unary, gather_scatter, qtensor, cumulative, mask, comparison, expand, bool, float, binary, int (all already covered in existing backend-tests) |

### What stays in flex

- Dtype storage paths (f16/bf16/f64/i8/i16/u8/u16/u32/u64) via direct `Flex::*` calls
- SIMD (row lengths 17/32) and rayon (rows ≥64/100/128) boundary probes
- Flex-internal helpers: `binary_op`, `scalar_op`, `broadcast_attn_mask_bias`, fused softmax/layer_norm, attention flash-vs-naive
- Validator panics, flip's zero-copy proof, expand's stride-metadata assertions
- Quantization scheme roundtrips + block-quant layout ops + f16/f64 dequantize
- FFT radix kernels, sort parallel threshold

Each remaining `mod tests` has a leading comment explaining what stays and pointing at the backend-tests file that covers the rest.

### Divergences surfaced

| Issue | Behavior | Gate |
|---|---|---|
| #4814 | NaN propagation in min/max/argmax/cummin/cummax. Flex+tch propagate (PyTorch/NumPy semantics); ndarray+cubecl drop NaN (IEEE 754). Public API docs are silent. | 7 tests gated `#[cfg(not(any(feature = "ndarray", feature = "cube")))]` |
| #4815 | Cubecl softmax reads underlying storage instead of materializing transposed view; returns softmax of pre-transpose rows. Flex uses `to_contiguous()` in `softmax_last`. | 1 test gated `#[cfg(not(feature = "cube"))]` |

Gate comments cite the tracking issue so unguarding is a one-line change once fixed.

### `burn-tensor` dev-dep

`burn-flex/src/` no longer imports `burn_tensor` (only a single doc-comment mention). The dev-dep stays in `Cargo.toml` because `burn-flex/benches/*.rs` still use the public `Tensor<B>` API.

**Benches will be moved to burn-backend-tests in a follow-up PR** — reusing its harness keeps what's measured aligned with real user workloads and lets each backend share benchmarks. Dropping the dev-dep falls out of that move.

### Incidental fixes

- Bumped `test_lu_*_block_dispatch` f16 tolerance from 5e-2 → 1.5e-1. Pre-existing flakiness: unseeded random 500-dim matrices occasionally overshot 5e-2 (observed 6.1e-2); still catches the O(1) regression from the f16 LU typo fixed in #4738.

### Test plan

| Suite | Result |
|---|---|
| `cargo test -p burn-backend-tests --features flex,std --test tensor` | 1304 passed · 10 ignored |
| `cargo test -p burn-backend-tests --features flex,std --test tensor_f16` | 901 passed · 6 ignored |
| `cargo test -p burn-backend-tests --features ndarray,std --test tensor` | 1490 passed · 16 ignored · NaN tests gated off |
| `cargo test -p burn-flex --lib` | 339 passed |
| `cargo check -p burn-flex --benches` | clean |

### CODE REVIEW

File: crates/burn-backend-tests/tests/tensor/bool/ops/expand.rs [hunk 1]
```diff
@@ -14,3 +14,15 @@ fn expand_2d_bool() {
...
 }
+
+#[test]
+fn expand_bool_after_flip() {
+    let tensor = TestTensorBool::<1>::from([true, false, true]).flip([0]);
+
+    let output = tensor.expand([2, 3]);
+
+    output.into_data().assert_eq(
+        &TensorData::from([[true, false, true], [true, false, true]]),
```
[the remark this answers is not in the document]
Comment (antimora) [reply crates/burn-backend-tests/tests/tensor/bool/ops/expand.rs]: Good catch — fixed in 1c22b5fdb. Switched to `[true, false, false]` which is asymmetric under flip, so the test now genuinely exercises the negative-stride path: flipped → `[false, false, true]` and the expected expanded values were updated to match.

**Review by laggui — APPROVED** [review]
Thanks for moving those 🙏 

Failing macos test is unrelated, I just filed an issue for that: #4828
